### ISSUE TITLE
Regression: OCaml language support fails completely after restarting language server

### ISSUE THREAD

**Issue by zepalmer** [ocamllabs/vscode-ocaml-platform#1261]
Greetings!  I'm currently running OCaml Platform v1.13.3 in VSCode 1.83.1 on a Debian Linux 12.2 system.  At present, I can only use VSCode for brief periods on average before OCaml language support fails entirely.  This appears to happen when new files are added to the project and under other circumstances I can't seem to identify.  That said, the problem is reproducible: I can trigger it immediately by opening VSCode and running the "OCaml: Restart Language Server" operation twice (although type hints disappear after running it once).  Doing so (or using VSCode long enough to trigger the problem in some other way) produces the following stack trace on the Output page for the OCaml Language Server tab:

```
Error: command 'ocamllsp/view-metrics' already exists
	at n.registerCommand (/usr/share/code/resources/app/out/vs/workbench/api/node/extensionHostProcess.js:125:82889)
	at Object.registerCommand (/usr/share/code/resources/app/out/vs/workbench/api/node/extensionHostProcess.js:140:20485)
	at ExecuteCommandFeature.register (/home/zpalmer/.vscode/extensions/ocamllabs.ocaml-platform-1.13.3/dist/vscode_ocaml_platform.bc.js:39:111862)
	at ExecuteCommandFeature.initialize (/home/zpalmer/.vscode/extensions/ocamllabs.ocaml-platform-1.13.3/dist/vscode_ocaml_platform.bc.js:39:111243)
	at LanguageClient.initializeFeatures (/home/zpalmer/.vscode/extensions/ocamllabs.ocaml-platform-1.13.3/dist/vscode_ocaml_platform.bc.js:44:8238)
	at LanguageClient.doInitialize (/home/zpalmer/.vscode/extensions/ocamllabs.ocaml-platform-1.13.3/dist/vscode_ocaml_platform.bc.js:41:10404)
	at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
	at async LanguageClient.start (/home/zpalmer/.vscode/extensions/ocamllabs.ocaml-platform-1.13.3/dist/vscode_ocaml_platform.bc.js:41:6304)
```

After this occurs, all language support other than syntax highlighting stops working.  Restarting the language server simply reproduces this error message.  Restarting the VSCode instance seems to be the only solution, after which everything will work for another brief period before returning to this state.  I have confirmed this behavior on a fresh installation (by moving `~/.vscode` out of the way) having only the OCaml Platform extension installed.

According to the [`registerCommand`](https://code.visualstudio.com/api/references/vscode-api#commands.registerCommand) documentation, this behavior seems to be expected if a command is registered with the identifier of an existing command.  Is it possible that the old `ocamllsp/view-metrics` command is not being unregistered before the language server is restarted?

I have no hypothesis for why this behavior regressed.

Having to close and reopen my editor multiple times per hour is pretty disruptive, so I'm quite invested in finding a solution to this problem.  Please let me know if there's anything I can do to help track this down!

**Comment by zepalmer** [comment]
For additional information: I have successfully reproduced the above behavior using the instructions found in the (`CONTRIBUTING.md`)[https://github.com/ocamllabs/vscode-ocaml-platform/blob/master/CONTRIBUTING.md] document (including the local OCaml 4.14.0 switch installed by the `Makefile`).  I was going to go bug hunting, but I'm not sure how to proceed as I'm not sure from where the `ocamllsp/view-metrics` command is coming; that string doesn't seem to appear in the `1.13.3` tag.

**Comment by zepalmer** [comment]
For further reference: the problem does not appear using OCaml Platform 1.13.1.  It first appears in OCaml Platform 1.13.2.  This data comes from using VSCode 1.83.1 and installing older versions of the extension from the marketplace via the extensions view.

**Comment by mnxn** [comment]
>  trigger it immediately by opening VSCode and running the "OCaml: Restart Language Server" operation twice

I can confirm that this reproduces the issue.

In version 1.13.2 of the extension, we updated our  `vscode-languageclient` dependency to 9.0.1.
The full stack trace below suggests it's either a problem with that library or how we're using it.

<details>
<summary>  Stack Trace </summary>
<pre>
2023-10-30 12:34:44.009 [error] TypeError: feature.clear is not a function
    at LanguageClient.cleanUp (/Users/max/.vscode/extensions/ocamllabs.ocaml-platform-1.13.3/node_modules/vscode-languageclient/lib/common/client.js:1010:21)
    at LanguageClient.shutdown (/Users/max/.vscode/extensions/ocamllabs.ocaml-platform-1.13.3/node_modules/vscode-languageclient/lib/common/client.js:968:14)
    at LanguageClient.stop (/Users/max/.vscode/extensions/ocamllabs.ocaml-platform-1.13.3/node_modules/vscode-languageclient/lib/common/client.js:935:21)
    at LanguageClient.client [as stop] (/Users/max/.vscode/extensions/ocamllabs.ocaml-platform-1.13.3/node_modules/vscode-languageclient/lib/node/main.js:150:22)
    at stop_server (/Users/max/.vscode/extensions/ocamllabs.ocaml-platform-1.13.3/_build/default/src/workspace_root/src/extension_instance.ml:29:11)
    at start_language_server (/Users/max/.vscode/extensions/ocamllabs.ocaml-platform-1.13.3/_build/default/src/workspace_root/src/extension_instance.ml:26:5)
    at f (/Users/max/.vscode/extensions/ocamllabs.ocaml-platform-1.13.3/_build/default/src/workspace_root/src/extension_instance.ml:35:3)
    at g (/Users/max/.vscode/extensions/ocamllabs.ocaml-platform-1.13.3/_build/default/src/builtin/+stdlib.js:41:16)
    at registerCommand (/Users/max/.vscode/extensions/ocamllabs.ocaml-platform-1.13.3/dist/vscode_ocaml_platform.bc.js:52:38577)
    at f (/Users/max/.vscode/extensions/ocamllabs.ocaml-platform-1.13.3/_build/default/src/workspace_root/src-bindings/vscode/vscode.ml:3212:5)
    at caml_callback (/Users/max/.vscode/extensions/ocamllabs.ocaml-platform-1.13.3/_build/default/src/builtin/+stdlib.js:28:12)
    at f (/Users/max/.vscode/extensions/ocamllabs.ocaml-platform-1.13.3/_build/default/src/builtin/+jslib.js:310:15)
    at /Users/max/.vscode/extensions/ocamllabs.ocaml-platform-1.13.3/_build/default/src/home/runner/work/vscode-ocaml-platform/vscode-ocaml-platform/_opam/lib/ojs/ojs_runtime.js:5:34
    at n.h (/Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/api/node/extensionHostProcess.js:125:84283)
    at n.$executeContributedCommand (/Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/api/node/extensionHostProcess.js:125:85143)
    at i.S (/Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/api/node/extensionHostProcess.js:134:10827)
    at i.Q (/Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/api/node/extensionHostProcess.js:134:10593)
    at i.M (/Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/api/node/extensionHostProcess.js:134:9683)
    at i.L (/Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/api/node/extensionHostProcess.js:134:8762)
    at s.value (/Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/api/node/extensionHostProcess.js:134:7565)
    at c.z (/Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/api/node/extensionHostProcess.js:80:1902)
    at c.fire (/Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/api/node/extensionHostProcess.js:80:2119)
    at m.fire (/Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/api/node/extensionHostProcess.js:96:14010)
    at s.value (/Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/api/node/extensionHostProcess.js:153:7902)
    at c.z (/Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/api/node/extensionHostProcess.js:80:1902)
    at c.fire (/Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/api/node/extensionHostProcess.js:80:2119)
    at m.fire (/Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/api/node/extensionHostProcess.js:96:14010)
    at MessagePortMain.<anonymous> (/Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/api/node/extensionHostProcess.js:153:6182)
    at MessagePortMain.emit (node:events:513:28)
    at Object.MessagePortMain._internalPort.emit (node:electron/js2c/utility_init:2:367)
2023-10-30 12:34:54.436 [error] TypeError: feature.clear is not a function
    at LanguageClient.cleanUp (/Users/max/.vscode/extensions/ocamllabs.ocaml-platform-1.13.3/node_modules/vscode-languageclient/lib/common/client.js:1010:21)
    at LanguageClient.shutdown (/Users/max/.vscode/extensions/ocamllabs.ocaml-platform-1.13.3/node_modules/vscode-languageclient/lib/common/client.js:968:14)
    at LanguageClient.stop (/Users/max/.vscode/extensions/ocamllabs.ocaml-platform-1.13.3/node_modules/vscode-languageclient/lib/common/client.js:935:21)
    at LanguageClient.stop (/Users/max/.vscode/extensions/ocamllabs.ocaml-platform-1.13.3/node_modules/vscode-languageclient/lib/node/main.js:150:22)
    at LanguageClient.doInitialize (/Users/max/.vscode/extensions/ocamllabs.ocaml-platform-1.13.3/node_modules/vscode-languageclient/lib/common/client.js:917:27)
    at processTicksAndRejections (node:internal/process/task_queues:95:5)
    at LanguageClient.start (/Users/max/.vscode/extensions/ocamllabs.ocaml-platform-1.13.3/node_modules/vscode-languageclient/lib/common/client.js:772:13)
2023-10-30 12:34:54.437 [error] Error: command 'ocamllsp/view-metrics' already exists
    at n.registerCommand (/Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/api/node/extensionHostProcess.js:125:82889)
    at Object.registerCommand (/Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/api/node/extensionHostProcess.js:140:20485)
    at ExecuteCommandFeature.register (/Users/max/.vscode/extensions/ocamllabs.ocaml-platform-1.13.3/node_modules/vscode-languageclient/lib/common/executeCommand.js:50:52)
    at ExecuteCommandFeature.initialize (/Users/max/.vscode/extensions/ocamllabs.ocaml-platform-1.13.3/node_modules/vscode-languageclient/lib/common/executeCommand.js:30:14)
    at LanguageClient.initializeFeatures (/Users/max/.vscode/extensions/ocamllabs.ocaml-platform-1.13.3/node_modules/vscode-languageclient/lib/common/client.js:1387:21)
    at LanguageClient.doInitialize (/Users/max/.vscode/extensions/ocamllabs.ocaml-platform-1.13.3/node_modules/vscode-languageclient/lib/common/client.js:890:18)
    at processTicksAndRejections (node:internal/process/task_queues:95:5)
    at LanguageClient.start (/Users/max/.vscode/extensions/ocamllabs.ocaml-platform-1.13.3/node_modules/vscode-languageclient/lib/common/client.js:772:13)
</pre>
</details>

**Comment by mnxn** [comment]
I believe I found the source of the issue. In https://github.com/microsoft/vscode-languageserver-node/commit/78995688495e4b0a35fe0a05377a0c8d5cad6518, the `dispose` field of `StaticFeature` is renamed to `clear`.

We need to update our languageclient bindings to reflect this change:

https://github.com/ocamllabs/vscode-ocaml-platform/blob/aad3850a2047459dd2a820b2b2a67130acee0ecf/src-bindings/vscode_languageclient/vscode_languageclient.mli#L162-L171

**Comment by smorimoto** [comment]
The fix for this is included in 1.13.4, which has just shipped:
https://github.com/ocamllabs/vscode-ocaml-platform/releases/tag/1.13.4

**Comment by zepalmer** [comment]
Thank you!
